### PR TITLE
Small comment and type correctness for Obj constructor

### PR DESCRIPTION
**Pull request by freider**
Just improves type + comment (got confused when I tried to refactor and this wasn't set)

### CODE REVIEW

**Comment by freider** [comment]
@prbot approve

**Comment by freider** [comment]
@prbot automerge

**Review by mwaskom — APPROVED** [review]

File: modal/cls.py [hunk 1]
```diff
@@ -90,7 +90,7 @@ def _uses_common_service_function(self):
 
     def __init__(
         self,
-        user_cls: type,
+        user_cls: Optional[type],  # this would be None in case of lookups
```
Comment (mwaskom) [inline modal/cls.py:93]: Just a random comment about consistency; I think the analogous attribute is `raw_f` for `Function`?

Comment (freider) [reply modal/cls.py:93]: Yup. There is a bunch of code here that needs cleanup/clarification. `FunctionInfo` is quite annoying how it bundles a bunch of different use cases but is hard to provide in some cases.

**Pull request merged**